ecosystem/node-checker/src/metric_evaluator/consensus/proposals_evaluator.rs

```diff
@@ -0,0 +1,214 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use super::{
+    super::{
+        common::{get_metric, GetMetricResult},
+        MetricsEvaluator, MetricsEvaluatorError,
+    },
+    CONSENSUS_EVALUATOR_SOURCE,
+};
+use crate::evaluator::EvaluationResult;
+use anyhow::Result;
+use clap::Parser;
+use log::debug;
+use poem_openapi::Object as PoemObject;
+use prometheus_parse::Scrape as PrometheusScrape;
+use serde::{Deserialize, Serialize};
+
+pub const CONSENSUS_PROPOSALS_EVALUATOR_NAME: &str =
+    const_format::concatcp!(CONSENSUS_EVALUATOR_SOURCE, "_", "proposals");
+
+// TODO: When we have it, switch to using a crate that unifies metric names.
+// As it is now, this metric name could change and we'd never catch it here
+// at compile time.
+const PROPOSALS_METRIC: &str = "aptos_consensus_proposals_count";
+
+#[derive(Clone, Debug, Deserialize, Parser, PoemObject, Serialize)]
+pub struct ConsensusProposalsEvaluatorArgs {}
+
+#[allow(dead_code)]
+#[derive(Debug)]
+pub struct ConsensusProposalsEvaluator {
+    args: ConsensusProposalsEvaluatorArgs,
+}
+
+impl ConsensusProposalsEvaluator {
+    pub fn new(args: ConsensusProposalsEvaluatorArgs) -> Self {
+        Self { args }
+    }
+
+    fn get_proposals_count(
+        &self,
+        metrics: &PrometheusScrape,
+        metrics_round: &str,
+    ) -> GetMetricResult {
+        let evaluation_on_missing_fn = || EvaluationResult {
+            headline: "Consensus proposals metric missing".to_string(),
+            score: 0,
+            explanation: format!(
+                "The {} set of metrics from the target node is missing the proposals metric: {}",
+                metrics_round, PROPOSALS_METRIC
+            ),
+            source: CONSENSUS_EVALUATOR_SOURCE.to_string(),
+            links: vec![],
+        };
+        get_metric(metrics, PROPOSALS_METRIC, None, evaluation_on_missing_fn)
+    }
+
+    fn build_evaluation(
+        &self,
+        previous_proposals_count: u64,
+        latest_proposals_count: u64,
+    ) -> EvaluationResult {
+        // We convert to i64 to avoid potential overflow if somehow the count somehow went backwards.
+        let progress = latest_proposals_count as i64 - previous_proposals_count as i64;
+        match progress {
```

Comment:
nit: the match seems completely unnecessary here. Why not just?
```
if progress == 0 {
...
} else if progress < 0 {
...
} else {
...
}
```

Reply:
I did this originally bc clippy told me to, but I agree this looks much better. I'll silence clippy in this case.

Reply:
Aah, if clippy is telling you to use the comparison operator, it probably wants something like this: https://users.rust-lang.org/t/greater-than-less-than-in-a-match-block/63399/5? If it's genuinely suggesting this logic, that sucks cause I think it's much uglier 😢 But, I'm also inclined not to silence clippy in our code and just do whatever it says. Otherwise, we'll have annotations everywhere.